### PR TITLE
[#91] Change Coin to Lovelace

### PR DESCRIPTION
**Pull request by ruhatch**
A simple name change that makes it clearer what a `Coin` is supposed to be.

### CODE REVIEW

**Review by JaredCorduan — APPROVED** [review]
LGTM, thanks!

File: src/Cardano/Chain/Common/AddrStakeDistribution.hs [hunk 1]
```diff
@@ -20,8 +20,8 @@ import qualified Formatting.Buildable as B (Buildable(..))
 
 import Cardano.Binary.Class (Bi(..), szCases)
 import qualified Cardano.Binary.Class as Bi
-import Cardano.Chain.Common.CoinPortion
-  (CoinPortion(..), coinPortionDenominator)
+import Cardano.Chain.Common.LovelacePortion
+  (LovelacePortion(..), lovelacePortionDenominator)
```
Comment (dnadales) [inline src/Cardano/Chain/Common/AddrStakeDistribution.hs:24]: Do we want to remove the `LovelacePortion` in the future as well?

File: src/Cardano/Chain/Common/Lovelace.hs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+-- | Constructor for 'Lovelace' returning 'LovelaceError' when @c@ exceeds 'maxLovelaceVal'
+mkLovelace :: Word64 -> Either LovelaceError Lovelace
+mkLovelace c
+  | c <= maxLovelaceVal = Right (Lovelace c)
+  | otherwise       = Left (LovelaceTooLarge (toInteger c))
+{-# INLINE mkLovelace #-}
+
+-- | Construct a 'Lovelace' from a 'KnownNat', known to be less than 'maxLovelaceVal'
+mkKnownLovelace :: forall n. (KnownNat n, n <= 45000000000000000) => Lovelace
```
Comment (dnadales) [inline src/Cardano/Chain/Common/Lovelace.hs:131]: Why not `45e15`?

File: src/Cardano/Chain/Common/Lovelace.hs [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+mkKnownLovelace = Lovelace . fromIntegral . natVal $ Proxy @n
+
+-- | Lovelace formatter which restricts type.
+lovelaceF :: Format r (Lovelace -> r)
+lovelaceF = build
+
+-- | Unwraps 'Lovelace'. It's called “unsafe” so that people wouldn't use it
+-- willy-nilly if they want to sum lovelace or something. It's actually safe.
+unsafeGetLovelace :: Lovelace -> Word64
```
Comment (dnadales) [inline src/Cardano/Chain/Common/Lovelace.hs:140]: hmmm, is this a function (and comment) we want to keep?

**Pull request merged**